.travis.yml

```diff
@@ -11,8 +11,13 @@ script:
 - bash -c 'time bin/bats --tap test'
 - |
   if [[ "$TRAVIS_OS_NAME" == 'linux' ]]; then
-    docker build --tag bats:latest .
-    docker run -it bats:latest --tap /opt/bats/test
+    RC=0
+    for bashver in 3.2 4.0 4.1 4.2 4.3 4.4
+    do
+      docker build --build-arg bashver=${bashver} --tag bats/bats:bash-${bashver} .
+      docker run -it bats/bats:bash-${bashver} --tap /opt/bats/test || RC=1
+    done
+    [ ${RC} -eq 0 ]
```

Comment:
I'm thinking it might be nice to capture which version failed, something like this:

```bash
failures=()
for bashver in ...
do
  ...
  if ! docker run -t bats/bats:bash-${bashver} --tap /opt/bats/test; then
    failures="$(docker run bash:${bashver} -c 'echo $BASH_VERSION')"
  fi
  if [[ "${#failures[@]}" -ne 0 ]]; then
    printf 'Bats failed under the following Bash versions:\n'
    printf -- '- %s\n' "${failures[@]}"
  fi
done
```

Thoughts? Make sense?

Reply:
Heh, realized I screwed up the first version of the code above, just edited it. Sorry for the spam.

Reply:
I see what you're going for and I like it.  I can only assume Travis uses Bash for underlying shell stuff since `[[` works.  I'll mess with that and see if I can get it implemented.